src/widgets/radio.jsx

```diff
@@ -574,17 +643,32 @@ var RadioEditor = React.createClass({
 
     onDelete: function(choiceIndex, e) {
         e.preventDefault();
+
         var choices = this.props.choices.slice();
+        var deleted = choices[choiceIndex];
+
         choices.splice(choiceIndex, 1);
-        this.props.onChange({choices: choices});
+
+        this.props.onChange({
+            choices: choices,
+            hasNoneOfTheAbove: !( deleted.isNoneOfTheAbove || !this.props.hasNoneOfTheAbove )
+        });
     },
 
-    addChoice: function(e) {
+    addChoice: function(noneOfTheAbove, e) {
         e.preventDefault();
 
         var choices = this.props.choices;
-        this.props.onChange({choices: choices.concat([{}])}, () => {
-            this.refs["editor" + choices.length].focus();
+        var newChoice = { isNoneOfTheAbove: !!noneOfTheAbove }
```

Comment:
I think this one was something I carried over when initially `noneOfTheAbove` could have been also `undefined`